plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/JavaLanguageModelBuilder.java

```diff
@@ -69,6 +73,37 @@ public DefaultJavaExtension getExtensionFor(Project project, SourceSet sourceSet
       extension.setCompilerArgs(compilerArgs);
       extension.setSourceCompatibility(getSourceCompatibility(compilerArgs));
       extension.setTargetCompatibility(getTargetCompatibility(compilerArgs));
+
+      // it's possible to set source/target compatibility in different places.
+      // Java plugin setting should override javacoptions
+      JavaVersion sourceVersion = null;
+      JavaVersion targetVersion = null;
+      if (GradleVersion.current().compareTo(GradleVersion.version("7.1")) >= 0) {
```

Comment:
What the difference between getting the source/target level from the java plugin and 

```
JavaCompile .getSourceCompatibility();
JavaCompile .getTargetCompatibility();
```
?

Reply:
I'd like to know too 😆 

The code managing it is [here](https://github.com/gradle/gradle/blob/dd40c5b0e06bef621941984ed8ba565dbec06805/platforms/jvm/plugins-java-base/src/main/java/org/gradle/api/plugins/JavaBasePlugin.java#L327)

I think that Gradle takes any source/target compatibility the user has set for the `java` plugin and applies it to the compiler task.  And then takes the compiler task settings and applies that to javacoptions.

But any of these can be set by the user in the build file and they can clash and Gradle seems fine with that.
They can also be illegal i.e. javacoptions require `target` >= `source` but it works out because the user has also specified `--release` in the options and that takes precedence.

I guess ultimately Gradle is just feeding all these settings into javacoptions but Intellij and Eclipse want the source/target compatibility for their own compilation settings and they seem to take that from the plugin or compiler settings rather than the javacoptions.

`Jabel` also seems to be taking advantage of this and using the source/target compatibility settings even though `--release` is set.

Reply:
I'll file a ticket at gradle for the recommended way to get the source/target level